src/hotspot/share/compiler/compilationPolicy.cpp

```diff
@@ -122,7 +122,7 @@ static inline CompLevel adjust_level_for_compilability_query(CompLevel comp_leve
 // Returns true if m is allowed to be compiled
 bool CompilationPolicy::can_be_compiled(const methodHandle& m, int comp_level) {
   // allow any levels for WhiteBox
-  assert(WhiteBoxAPI || comp_level == CompLevel_any || is_compile(comp_level), "illegal compilation level");
+  assert(WhiteBoxAPI || comp_level == CompLevel_any || comp_level == CompLevel_all || is_compile(comp_level), "illegal compilation level");
```

Comment:
@JohnTortugo could you please review this? I assume you have audited all other places where we use `CompLevel_any` and `CompLevel_all `.

Reply:
Looking at other usages of `CompLevel_any`/`CompLevel_all`, I'm concerned that JDK-8334046 accidentally introduced some semantic changes. I think we should back it out and redo.